gallery/visualtests/nxTag.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+const { clickTest, focusTest, focusAndHoverTest, hoverTest, simpleTest } = require('./testUtils');
+const { Region, Target } = require('@applitools/eyes-webdriverio');
+
+describe('NxTag', function() {
+  beforeEach(async function() {
+    await browser.url('#/pages/NxTag');
+  });
+
+  describe('Basic NxTag', function() {
+    const selector = '#nx-tag-example .gallery-example-live';
+
+    it('looks right', simpleTest(selector));
+  });
+
+  describe('NxTag with overflow has a tooltip', function() {
+    const selector = '.gallery-example-live',
+          tagSelector = `${selector} .nx-tag--pink`,
+
+          // expected distance from top of element to the top of its tooltip
+          tooltipHeightOffset = 45;
+
+    it('looks right', async function() {
+      const [tagElement, tooltipTagElement] = await Promise.all([browser.$(selector), browser.$(tagSelector)]);
+
+      await tagElement.scrollIntoView({ block: 'center' });
+
+      const { x, y, height, width } = await browser.getElementRect(tagElement.elementId);
+
+      // hover the tag to activate its tooltip.
+      await tooltipTagElement.moveTo();
+
+      const screenshotRegion = new Region(x, y - tooltipHeightOffset, width, height + tooltipHeightOffset);
```

Comment:
This only captures the part of the tooltip that is directly above the tag, right?  I guess that's good enough for proving it's there

Reply:
> This only captures the part of the tooltip that is directly above the tag, right? I guess that's good enough for proving it's there

The screenshot covers all of the tags plus the tooltip. It looks good to me.

Reply:
> The screenshot covers all of the tags plus the tooltip. It looks good to me.

Ah I see, because you're using the `.gallery-example-live` as the basis for your screenshot box.  In that case, be aware that your `tooltipHeightOffset` is much higher than needed, since the tooltip is attached to a tag, not to the `.gallery-example-live`.  You can subtract 24 (the amount of padding on `.gallery-example-live` from `tooltipHeightOffset`.

Reply:
> > The screenshot covers all of the tags plus the tooltip. It looks good to me.
> 
> Ah I see, because you're using the `.gallery-example-live` as the basis for your screenshot box. In that case, be aware that your `tooltipHeightOffset` is much higher than needed, since the tooltip is attached to a tag, not to the `.gallery-example-live`. You can subtract 24 (the amount of padding on `.gallery-example-live`) from `tooltipHeightOffset`.

https://github.com/sonatype/sonatype-react-shared-components/pull/290/commits/6b265bf3b5ac145e8abcea32ba9cf0513dd278a8